packages/js/src/dashboard/services/widget-factory.js

```diff
@@ -67,6 +68,9 @@ export class WidgetFactory {
 					remoteDataProvider={ this.#remoteDataProvider }
 				/>;
 			case WidgetFactory.types.topPages:
+				if ( ! this.#dataProvider.getSiteKitConfiguration().isConnected ) {
```

Comment:
once connected and the dashboard re-renders, we show the widget. No need to add the widget in the site kit setup widget. We only need to re render the dashboard.